pkg/chunked/storage_linux.go

```diff
@@ -1987,6 +2017,10 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions, diff
 
 			switch chunk.ChunkType {
 			case internal.ChunkTypeData:
+				fm := fileMetadata{
+					FileMetadata: *chunk,
+				}
+				chunk := &fm
```

Comment:
Following the principle that `fileMetadata` entries only contain `TypeChunk` entries in the `.Chunks` field, not at the top level, it seems to me that the type of the `findChunkInOtherLayers` parameter could remain `internal.FileMetadata`, so that we don’t need to create this usually-invalid entry.